modules/apps/commerce/commerce-product-definitions-web/src/main/resources/META-INF/resources/definition_links.jsp

```diff
@@ -95,7 +95,7 @@ long cpDefinitionId = cpDefinitionLinkDisplayContext.getCPDefinitionId();
 							}
 						},
 						title:
-							'<liferay-ui:message arguments="<%= HtmlUtil.escapeJS(cpDefinition.getName(languageId)) %>" key="add-new-product-to-x" />',
+							'<liferay-ui:message arguments="<%= cpDefinition.getName(languageId) %>" key="add-new-product-to-x" />',
```

Comment:
Is the modal itself that escapes this? Otherwise where is the "second escape"?

Reply:
Hi @smottal, it's the [messageTag](https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/util-taglib/src/com/liferay/taglib/ui/MessageTag.java#L76) itself when formatting to Unicode. So the escapeJS is unnecessary.

Reply:
this escape was introduced by this https://liferay.atlassian.net/browse/COMMERCE-5017. I'm not sure this is the right fix.

Reply:
After analysing this I agree this is the right solution